src/preview/VideoPreview.test.tsx

```diff
@@ -0,0 +1,190 @@
+		await user.click(action1Item);
+		expect(actions[0].onClick).toHaveBeenCalled();
+		expect(onClose).toHaveBeenCalledTimes(1);
+		// click on action 2 skips the handler of the action since it is disabled and calls onClose
+		await user.click(action2Item);
+		expect(actions[1].onClick).not.toHaveBeenCalled();
+		expect(onClose).toHaveBeenCalledTimes(1);
+		// click on close action is the onClose itself
+		await user.click(closeActionItem);
+		expect(onClose).toHaveBeenCalledTimes(2);
+		// click on filename is equivalent to a click on the overlay, so onClose is called
+		await user.click(screen.getByText(/video name/i));
+		expect(onClose).toHaveBeenCalledTimes(3);
+	});
+
+	it('should not render the video when canPlayType return empty string on mime type(mime type not supported)', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('');
+
+		setup(
+			<VideoPreview
+				onClose={jest.fn()}
+				show
+				src={''}
+				filename="video name"
+				mimeType={'unsupported/mimeType'}
+			/>
+		);
+
+		expect(screen.getByText(cannotBePlayedMessage)).toBeVisible();
+		expect(screen.queryByTestId('video')).not.toBeInTheDocument();
+	});
+
+	it('should render the video when mime type props is not provided', () => {
+		setup(<VideoPreview onClose={jest.fn()} show src={''} filename="video name" />);
+
+		expect(screen.getByTestId('video')).toBeVisible();
+		expect(screen.queryByText(cannotBePlayedMessage)).not.toBeInTheDocument();
+	});
+
+	it('should render the video when canPlayType return maybe string on mime type', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('maybe');
+		const mimeType = 'video/mp4';
+		setup(<VideoPreview onClose={jest.fn()} show src={''} mimeType={mimeType} />);
+		expect(screen.getByTestId('video')).toBeVisible();
+		expect(screen.queryByText(cannotBePlayedMessage)).not.toBeInTheDocument();
+	});
+
+	it('should render the video when canPlayType return probably string on mime type', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('probably');
+		const mimeType = 'video/mp4';
+		setup(<VideoPreview onClose={jest.fn()} show src={''} mimeType={mimeType} />);
+		expect(screen.getByTestId('video')).toBeVisible();
+		expect(screen.queryByText(cannotBePlayedMessage)).not.toBeInTheDocument();
+	});
+
+	it('should render the fail string when video request fails', async () => {
+		setup(<VideoPreview onClose={jest.fn()} show src={''} />);
+		fireEvent.error(screen.getByTestId('video'));
+		expect(await screen.findByText(cannotBePlayedMessage)).toBeVisible();
+	});
+
+	it('should call video play when keyboard space is clicked', async () => {
+		jest.spyOn(HTMLMediaElement.prototype, 'paused', 'get').mockReturnValue(true);
+		const playStub = jest
+			.spyOn(window.HTMLVideoElement.prototype, 'play')
+			.mockImplementation(() => Promise.resolve());
+
+		const pauseStub = jest
+			.spyOn(window.HTMLVideoElement.prototype, 'pause')
+			.mockImplementation(() => undefined);
```

Comment:
The implementation is not useful
```suggestion
			.mockImplementation();
```